internal/util/rand.1.22.go

```diff
@@ -0,0 +1,29 @@
+//go:build go1.22
+// +build go1.22
+
+package util
+
+import (
+	"encoding/binary"
+	"math/rand/v2"
+)
+
+func Shuffle(n int, swap func(i, j int)) {
+	rand.Shuffle(n, swap)
+}
+
+func IntN(n int) int {
```

Comment:
until.IntN looks vague. I think we can just replace it with the below FastRand.

Reply:
We can do that,  but for go !1.22 
The regular usage of Intn would also be replaced with rngPool implementation, Should we do that? @rueian

Reply:
Yes, that is okay.